### PR TITLE
docs: updated collapse props. added example for using the config prop

### LINKED ISSUE #1017: [Collapse] Duration prop is not applying
**Describe the bug** No matter what I do, changing the `duration` value on a `<Collapse />` component doesn't seem to have any effect

**Expected Behavior** ... it should have an effect

**Link to minimal reproduction** https://codesandbox.io/s/eloquent-lumiere-0glyc?fontsize=14&hidenavigation=1&theme=dark

**Steps to Reproduce**:
1. Create a `<Collapse>` and fill it with a lot of content
2. Add a `duration` prop and set to some high value, say 10 seconds (`10000`)

**Suggested solution(s)** N/A

**Desktop (please complete the following information):**

- OS: Mac OS 10.15.5
- Browser: Chrome 83

**Additional context** N/A

### PR DESCRIPTION
**Pull request by jmiazga**
Added an example of the `config` prop to the Collapse docs. Also updated the props chart.

Fixes #1017

### CODE REVIEW

File: website/docs/components/collapse.mdx [hunk 1]
```diff
@@ -72,16 +72,58 @@ function Example() {
...
 
+### Customizing the transition state styles
+
+The `config` prop is used to style the Transition component's initial, entered, and 
+exiting states. This allows the height, opacity, transform, transition, or any 
+other valid CSS property to be configured for each state.
+
+```jsx
+function Example() {
+  const [show, setShow] = React.useState(false)
```
Comment (with-heart) [inline website/docs/components/collapse.mdx:83]: Could you replace this and the related functionality with `useDisclosure`?

**Comment by segunadebayo** [comment]
I'd like to review this later tomorrow before merging.

File: website/docs/components/collapse.mdx (hunk 1 above)
Comment (jmiazga) [reply website/docs/components/collapse.mdx]: Yep updated every example in the Collapse docs.

**Comment by segunadebayo** [comment]
Thanks @jmiazga for this PR.

I've updated the `Collapse` component as mentioned here #1017. So this fix won't be needed.

I appreciate you for taking the time to fix it.

**Pull request closed without merging**